### ISSUE TITLE
Not getting correct error message from Google signin.

### ISSUE THREAD

**Issue by AmbrishPatel94** [aws-amplify/amplify-swift#3176]
### Describe the bug

I am use following Amplify version 2.0.0-Next Major, I register new user on amplify and try to login with amplify google signin "Amplify.Auth.signInWithWebUI(for: .google, presentationAnchor: presentationAnchor, options: .preferPrivateSession())" with same user then that time i got error response "invaid_request". But i need to get response like this "invalid_request: PreSignUp failed with error {"error":{"code":409,"custom_code":"002","message":"U2FsdGVkX19xJYJt/Dckptj0KKXuBfrD3uhZScr1bBvqYF2Q29iGb7iC16MDCwLwbIVs1tNcaHQreEIopKwg9onymeCveyiCwHNzjpmPciQ3XSiQ7Oi+pVW4KyB5lUXC"}}."

### Steps To Reproduce

```swift
I register new user on amplify and try to login with amplify google signin "Amplify.Auth.signInWithWebUI(for: .google, presentationAnchor: presentationAnchor, options: .preferPrivateSession())" with same user then that time i got error response "invaid_request".
```

### Expected behavior

I need to get response like this "invalid_request: PreSignUp failed with error {"error":{"code":409,"custom_code":"002","message":"U2FsdGVkX19xJYJt/Dckptj0KKXuBfrD3uhZScr1bBvqYF2Q29iGb7iC16MDCwLwbIVs1tNcaHQreEIopKwg9onymeCveyiCwHNzjpmPciQ3XSiQ7Oi+pVW4KyB5lUXC"}}."

### Amplify Framework Version

2.0.0-Next Major

### Amplify Categories

Auth

### Dependency manager

Swift PM

### Swift version

5

### CLI version

10.6.1

### Xcode version

14.2

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


INSERT LOG MESSAGES HERE
```
</details>
```


### Is this a regression?

Yes

### Regression additional context

_No response_

### Platforms

iOS

### OS Version

13.0

### Device

iphone 13

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by atierian** [comment]
Hey @AmbrishPatel94, thanks for opening this issue.
If I understand correctly, you're receiving an error with only `invalid_request` but are expecting an error with more information like in the example you provided:
```
"invalid_request: PreSignUp failed with error {"error":{"code":409,"custom_code":"002","message":"U2FsdGVkX19xJYJt/Dckptj0KKXuBfrD3uhZScr1bBvqYF2Q29iGb7iC16MDCwLwbIVs1tNcaHQreEIopKwg9onymeCveyiCwHNzjpmPciQ3XSiQ7Oi+pVW4KyB5lUXC"}}."
```

Can you please provide a full code example of how you're catching that error?

**Comment by AmbrishPatel94** [comment]
Hi Atierian, thanks for reaching out us. 

I am using this 
 Task{
            do {
                let signInResult = try await Amplify.Auth.signInWithWebUI(for: .google, presentationAnchor: self.view.window!, options: .preferPrivateSession())
                
                if signInResult.isSignedIn {
                    print("Sign in succeeded")
                }else{
                    
                }
            } catch let error as AuthError {
                print("Sign in failed \(error )")
            // here i am receiving error message invalid_request

            } catch {

                print("Unexpected error: \(error)")


            }
        }

**Comment by AmbrishPatel94** [comment]
> Hi Atierian, thanks for reaching out us.
> 
> I am using this Task{ do { let signInResult = try await Amplify.Auth.signInWithWebUI(for: .google, presentationAnchor: self.view.window!, options: .preferPrivateSession())
> 
> ```
>             if signInResult.isSignedIn {
>                 print("Sign in succeeded")
>             }else{
>                 
>             }
>         } catch let error as AuthError {
>             print("Sign in failed \(error )")
>         // here i am receiving error message invalid_request
> 
>         } catch {
> 
>             print("Unexpected error: \(error)")
> 
> 
>         }
>     }
> ```

**Comment by AmbrishPatel94** [comment]
Hi Atierian,

I am waiting to your response.

**Comment by atierian** [comment]
It looks like you accidentally closed the issue, so I'm reopening it.

Thanks for the callsite example. You mentioned that you're receiving the error message in this `catch let error as AuthError` block. Please paste the full content of the error message printed there. Based on the code block you provided, there should be more that just `invalid_request`.
```swift
catch let error as AuthError {
    print("Sign in failed \(error )")
    // here i am receiving error message invalid_request
}
```

**Comment by AmbrishPatel94** [comment]
I am using Amplify version 2.15.4 yes we are using a PreSignUp_ExternalProvider lambda trigger here.

**Comment by atierian** [comment]
Thanks. 

The error that you're expecting to receive:
```
invalid_request: PreSignUp failed with error {"error":{"code":409,"custom_code":"002","message":"U2FsdGVkX19xJYJt/Dckptj0KKXuBfrD3uhZScr1bBvqYF2Q29iGb7iC16MDCwLwbIVs1tNcaHQreEIopKwg9onymeCveyiCwHNzjpmPciQ3XSiQ7Oi+pVW4KyB5lUXC"}}.
```
Are you providing this whole error to Cognito in your `PreSignUp_ExternalProvider` triggered lambda?

If so, can you please paste the error handling portion of the lambda here?

**Comment by AmbrishPatel94** [comment]
We are using this for PreSignUp_ExternalProvider

`if (event.triggerSource === "PreSignUp_ExternalProvider") {
                console.log(
                    `Identified :: This call is for external service provider. But, its duplicated flow so we need to stop account creation & send error message 409 with 002.`
                );

                // If count of duplicationData is 1 then we link the accounts & stop account creation with 409 / 002 else we stop account creation with 409 / 001 so chooser window can be shown on FE
                if (duplicationData.Users.length === 1) {
                    console.log(
                        `The count is exactly 1 so we need to link the account & create 409 with 002 situation.`
                    );

                    let attributes = duplicationData.Users[0].Attributes;
                    let GUID = attributes.find(
                        (attribute) => attribute["Name"] === "sub"
                    )["Value"];

                    console.log(
                        `This is the GUID from attributes - ${GUID} | Now, Attempting to link the provider account :: `
                    );

                    await linkProviderAccount(
                        duplicationData.Users[0].Username,
                        event.userName
                    );
                    console.log(`Now, Attempting to update the email_verified : true inside the cognito :: `);
                    await updateUserInCognito(duplicationData.Users[0].Username);
                    console.log(
                        `Now, Attempting to generate the authcode using the generateAuthCode fn from the layer :: `
                    );
                    let message = await generateAuthCode(GUID);

                    console.log(`Now, Attempting to encrypt the authcode :: `);
                    let AuthToken = await encrypt(
                        Buffer.from(message, "utf-8")
                    );

                    console.log(
                        `Now, Attempting to store the authcode in dynamoDB :: `
                    );
                    await storeAuthToken(GUID, AuthToken);

                    // Importing the checkUsernameType function from helper in this file & sending this as a param to this function so it can be used there as well.
                    let duplicateAccounts = await generateDuplicateAccounts(
                        checkUsernameType,
                        duplicationData
                    );
                    let encryptedDuplicateAccounts = await encryptData(
                        JSON.stringify(duplicateAccounts)
                    );

                    // Passwordless sign-in attempt
                    let errorResponseData = await errorResponse(
                        "409",
                        "002",
                        encryptedDuplicateAccounts
                    );
                    callback(new Error(errorResponseData), event);

                    console.log(
                        `/Safely exiting the scenario :: 409 with 002 | Stopping the account creation`
                    );
                    return;
                }`

**Comment by atierian** [comment]
Great, thanks.
It looks like this is because we're extracting the `error` query parameter, but not the `error_description`.

Could you try out the branch `hostedui_errordescription` and see if that resolves your issue?

**Comment by AmbrishPatel94** [comment]
HI Atierian,
Thanks for get solution.
But my question is, How can i use this code because i a using this sdk by Package dependency and there are no option to edit.
Can you add this code on new version?

**Comment by atierian** [comment]
This fix was included in Amplify Swift v2.16.1
If you're still using the version rule `Up to Next Major 2.0.0`, you can select `File > Packages > Update to Latest Package Versions` to update to `2.16.1`. 

Please let us know whether this new version resolves your issue. Thanks again for reporting it!
